sklearn/tests/test_kernel_approximation.py

```diff
@@ -332,3 +332,31 @@ def test_nystroem_precomputed_kernel():
                       **param)
         with pytest.raises(ValueError, match=msg):
             ny.fit(K)
+
+
+def test_landmarks_using():
+    # Basic tests to check the work of the landmarks selection.
+    rnd = np.random.RandomState(0)
+    X = rnd.uniform(size=(10, 4))
+    # test that as a result of the fit we really get the components that we
+    # originally chose
+    landmarks = rnd.permutation(X.shape[0])
+    for i in range(1, len(landmarks)):
+        nystroem = Nystroem(n_components=i, kernel=_linear_kernel,
+                            random_state=rnd)
+        nystroem.fit(X, landmarks=landmarks[:i])
+        assert_array_almost_equal(X[landmarks[:i]], nystroem.components_)
+    #test that n_components reduced to the size of landmarks
+    landmarks = rnd.permutation(X.shape[0]//3)
+    n_components = 5
+    nystroem = Nystroem(n_components=n_components, kernel=_linear_kernel,
+                        random_state=rnd)
+    nystroem.fit(X, landmarks=landmarks)
+    assert_array_equal(landmarks, nystroem.component_indices_)
+    # test correctly returned error in case the landmarks are out of range
+    msg = 'landmarks indices out of X range'
```

Comment:
```suggestion
    # test raised error in case the landmarks are out of range
    msg = f"The following landmarks are outside of the range of X: {X.shape[0]}"
```